pkg/guid/guid_pool.go

```diff
@@ -19,8 +20,14 @@ type Pool interface {
 	// ReleaseGUID release the reservation of the guid.
 	// It returns error if the guid is not in the range.
 	ReleaseGUID(string) error
+
+	// Reset clears the current pool and resets it with given values (may be empty)
+	Reset(guids []string) error
 }
 
+var GuidPoolExhaustedError = errors.New("GUID pool is exhausted")
```

Comment:
Should it be const?

Reply:
`errors.New` doesn't yield a const expression, so we can't do that :(